addons/hr_holidays/models/hr_leave.py

```diff
@@ -111,9 +111,120 @@ def _default_employee(self):
         ('type_value', "CHECK( (holiday_type='employee' AND employee_id IS NOT NULL) or (holiday_type='category' AND category_id IS NOT NULL) or (holiday_type='department' AND department_id IS NOT NULL) )",
          "The employee, department or employee category of this request is missing. Please make sure that your user login is linked to an employee."),
         ('date_check2', "CHECK ((date_from <= date_to))", "The start date must be anterior to the end date."),
-        ('date_check', "CHECK ( number_of_days_temp >= 0 )", "The number of days must be greater than 0."),
+        ('date_check', "CHECK ( number_of_days_temp >= 0 )", "If you want to change the number of days you should use the 'period' mode"),
     ]
 
+    # Those fields are mostly for the interface only
+
+    display_date_from = fields.Date()
+    display_date_to = fields.Date()
+
+    method = fields.Selection(related='holiday_status_id.method', readonly=True)
+
+    date_from_period = fields.Selection([('am', 'Morning'), ('pm', 'Afternoon')], string="Date Period Start", default='am')
+    date_to_period = fields.Selection([('am', 'Morning'), ('pm', 'Afternoon')], string="Date Period End", default='pm')
```

Comment:
request_date_to_period ? + probably add an help telling it is used when computing leaves based on half-day computation